src/dbus.cpp

```diff
@@ -657,37 +688,35 @@ void dbus_manager::start_thread()
 {
     stop_thread();
     m_quit = false;
-    m_thread = std::thread(dbus_thread, this);
+    m_thread = std::thread(&dbus_manager::dbus_thread, this);
 }
 
-void dbus_manager::dbus_thread(dbus_manager *pmgr)
+void dbus_manager::dbus_thread()
 {
     (void)parse_property_changed;
     DBusMessage *msg = nullptr;
-    auto& dbus = pmgr->dbus();
-
-    const std::string main_dest = "org.mpris.MediaPlayer2." + pmgr->m_dest;
 
     // loop listening for signals being emmitted
-    while (!pmgr->m_quit) {
+    while (!m_quit) {
 
         // non blocking read of the next available message
-        if (!dbus.connection_read_write(pmgr->m_dbus_conn, 0))
+        if (!m_dbus_ldr.connection_read_write(m_dbus_conn, 0))
             return; // connection closed
 
-        msg = dbus.connection_pop_message(pmgr->m_dbus_conn);
+        msg = m_dbus_ldr.connection_pop_message(m_dbus_conn);
 
         // loop again if we haven't read a message
         if (nullptr == msg) {
-            std::this_thread::sleep_for(ms(10));
+            std::this_thread::yield();
```

Comment:
dbus.connection_read_write() should actually have a timeout set then?

Reply:
It should in this case, because the thread is doing nothing but dbus interaction. I tested it and it seemed to block the main thread. Until I figure out why, I left it like this